src/python/pants/option/native_options.py

```diff
@@ -67,7 +97,46 @@ def __init__(
             (dict, None): self._native_parser.get_dict,
         }
 
-    def get(
+    def with_derivation(self) -> NativeOptionParser:
+        """Return a clone of this object but with value derivation enabled."""
+        # We may be able to get rid of this method once we remove the legacy parser entirely.
+        # For now it is convenient to allow the help mechanism to get derivations via the
+        # existing Options object, which otherwise does not need derivations.
+        return NativeOptionParser(
+            args=None if self._args is None else tuple(self._args),
+            env=dict(self._env),
+            config_sources=None if self._config_sources is None else tuple(self._config_sources),
+            allow_pantsrc=self._allow_pantsrc,
+            include_derivation=True,
+        )
+
+    def get_value(self, *, scope, registration_args, registration_kwargs) -> Tuple[Any, Rank]:
```

Comment:
It seems slightly unfortunate/error-prone to have no type hints on these args. Did you think about this explicitly? If so, can you explain your thinking? Thanks.

Reply:
Well `scope` is a str, no problem annotating that. But the other two are `*args` and `**kwargs` as captured at registration time, and so can be ~anything. Well, kwargs is a dict of str to anything, but that is not much help either.

We mostly don't annotate them in the legacy parser either. I see one place where we assumed that the args were all `str`s, but that's not actually correct... So I figure no annotation is better an a broken one.

Reply:
FWIW, once the legacy parser is gone I hope to streamline this and not have to pass these args/kwargs around outside the class.

Reply:
Ah, yeah, I see. Makes sense! Thanks for explaining.

(No change required from my perspective)